solr/solrj/src/test/org/apache/solr/client/solrj/impl/CloudHttp2SolrClientTest.java

```diff
@@ -291,7 +297,7 @@ public void testHttpCspPerf() throws Exception {
       // lookup)
       assertLogCount(collectionClusterStateLogs, 1);
       // 1 call to fetch entire cluster state from HttpCSP.fetchLiveNodes()
-      assertLogCount(entireClusterStateLogs, 1);
+      assertLogCount(entireClusterStateLogs, 0);
```

Comment:
No more calls to fetch entire cluster state!

Reply:
Awesome.

At this juncture, I recommend changes to this test because the way it works now is awkward:  Only have one LogListener for the "adminRequestLogs" (simple; 3 is more confusing).  Consume the log messages in the test (don't merely count) to assert that the log message is what it should be, ideally with a substring check or a simple regex if you must.  Consume & assert this after each step that we know will emit a request:  after client construction and once more after the first "add" (doc).  Your code comments here are then redundant as the assertions will be more specific at the right times to be self explanatory.

Reply:
I agree. Improved the test